bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -619,7 +619,10 @@ std::pair<PrePrepareMsg *, bool> ReplicaImp::buildPrePrepareMessage() {
   TimeRecorder scoped_timer(*histograms_.buildPrePrepareMessage);
   PrePrepareMsg *prePrepareMsg = createPrePrepareMessage();
   if (!prePrepareMsg) return std::make_pair(nullptr, false);
-  SCOPED_MDC("pp_msg_cid", prePrepareMsg->getCid());
+
+  if (!getReplicaConfig().prePrepareFinalizeAsyncEnabled) {
+    SCOPED_MDC("pp_msg_cid", prePrepareMsg->getCid());
```

Comment:
In case prePrepareFinalizeAsyncEnabled is false, what CID will appear in the logs now?

Reply:
if prePrepareFinalizeAsyncEnabled is false then preprepare will be created in single thread. So the correct CID will appear in logs. If prePrepareFinalizeAsyncEnabled is true then the CID will be added only at the time of consensus, since at the time of building the preprepare, the CID allocated might change later.